share/arno-iptables-firewall/environment

```diff
@@ -1614,26 +1614,39 @@ check_interface()
   return 1
 }
 
-# Get IP address/mask of specified network interface
+# Get all IP address(es)/mask(s) of specified network interface
+get_network_ipv4_address_mask_all()
+{
+  ip -o addr show dev "$1" 2>/dev/null \
+    |awk '$3 == "inet" { print $4 }' |tr '\n' ' '
+}
+
+# Get (primary) IP address/mask of specified network interface
 get_network_ipv4_address_mask()
+{
+  get_network_ipv4_address_mask_all "$1" |cut -d' ' -f1
```

Comment:
While this appears to work, using `cut` with a space delimiter will fail if any extra space would appear in the output.

You could have skipped the `|tr '\n' ' '` and used `head -n1` to get the first line, or in a `for` loop use `unset IFS` to include `\n` as whitespace.

That said, these environment changes seem to work to my eye.

Reply:
I know cut is bad with multi-spaces but since get_network_ipv4_address_mask_all is implemented by us and not a system function, we *know* that it uses one space. For other stuff I normally use awk -F' ' '{ print $1 }' but the 'cut' solution is cleaner imo.

And you're referring to the |tr '\n' ' 'in get_network_ipv4_address_mask_all, right? I'm not sure what you mean, since that's exactly the idea: it should use space as a separator when multiple IP addresses are returned.